docs/pages/core_lib_doc/Testing Core-lib/test_core_lib.md

```diff
@@ -0,0 +1,242 @@
+---
+id: test_core_lib
+title: Testing Core Lib
+sidebar: core_lib_doc_sidebar
+permalink: test_core_lib.html
+folder: core_lib_doc
+toc: false
+---
+
+Let's understand how `Core-Lib` is initialized and tested and how to integrate it with your new or existing application.
+
```

Comment:
We need to add another case where we are using the same code lib. but have multiple test files. and we want to use the same CoreLib instance with this test. so


```

threadLock = threading.Lock()


class OblInstange(object):
    instance = None
    config = None


def load_config():
    if not OblInstange.config:
        path = os.path.join(os.path.dirname(__file__), '..', 'data')
        load_dotenv(dotenv_path=os.path.join(path, '.env'))

        GlobalHydra.instance().clear()
        hydra.initialize(config_path=os.path.join('..', 'data', 'config'), caller_stack_depth=1)
        OblInstange.config = hydra.compose('config.yaml')
    return OblInstange.config


def sync_create_start_core_lib() -> ObjectiveLoveCoreLib:
    threadLock.acquire()
    if not OblInstange.instance:
        try:
            [CoreLib.cache_registry.unregister(key) for key in CoreLib.cache_registry.registered()]
            [CoreLib.observer_registry.unregister(key) for key in CoreLib.observer_registry.registered()]
            OblInstange.instance = ObjectiveLoveCoreLib(load_config())
            # ob_love_core_lib._messaging_client = patch('objective_love_core_lib.client.messaging_client.MessagingClient')
            OblInstange.instance.start_core_lib()
        except BaseException as e:
            print(''.join(traceback.format_exception(type(e), e, e.__traceback__)))
            raise e

    # Clear the cache
    for key in CoreLib.cache_registry.registered():
        CoreLib.cache_registry.get(key).flush_all()


    threadLock.release()
    return OblInstange.instance

```